api/envoy/type/v3/ratelimit_strategy.proto

```diff
@@ -0,0 +1,82 @@
+syntax = "proto3";
+
+package envoy.type.v3;
+
+import "envoy/type/v3/ratelimit_unit.proto";
+import "envoy/type/v3/token_bucket.proto";
+
+import "xds/annotations/v3/status.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.type.v3";
+option java_outer_classname = "RatelimitStrategyProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/type/v3;typev3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+option (xds.annotations.v3.file_status).work_in_progress = true;
+
+// [#not-implemented-hide:]
+// [#protodoc-title: Rate Limit Strategies]
+
+message RateLimitStrategy {
+  // Choose between allow all and deny all.
+  enum BooleanChoice {
```

Comment:
Good point. Initially I made it two separate messages, and then reduced them to an enum. I think we can drop this enum for good, and just use `RequestsPerTimeUnit`.

Reply:
Ah, I remember why I made it a separate message/strategy. This is used when the only strategy allowed is to deny/allow, f.e. RateLimitStrategyQueue

```protobuf
// Whether to deny the requests discarded due to queue lengths limit,
// or allow them.
RateLimitStrategy.BooleanChoice spillover_discard_strategy = 3;
```

and `NoAssignmentBehavior`:

```protobuf
// Allow or Deny the requests until the bucket is assigned a quota.
type.v3.RateLimitStrategy.BooleanChoice process_requests = 1;
```

Reply:
I do agree `BooleanChoice` is a bad name, I really struggled naming this one.

Reply:
I ended up renaming it to `BlanketRule` in a96f81998. I kept it as an enum, so that it's clear that ALLOW_ALL is the default value. Let me know if this works.

Reply:
I guess that seems reasonable.  It's a little cumbersome here, but I guess the alternatives are a little cumbersome as well.  So this is fine.